app/services/parse_csv_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'logger'
+
+module Services
+  class ParseCsvService
+    def initialize(csv_data)
+      @csv_data = csv_data
+    end
+
+    def call
+      rows = CSV.parse(@csv_data, headers: false)
+      rows.map do |row|
+        validate_row(row)
```

Comment:
is this a reasonable expectation for the end user: to get 0 results if the csv rows are malformed?

Reply:
Yeah, it should throw an error if all rows are malformed otherwise we should use the ok rows and throw specific errors for the other ones.